src/File.ts

```diff
@@ -125,10 +132,20 @@ const tryRepetitive = async ({
         }
 
         const line = fileLines[listItemCache.position.start.line];
-
         if (line.includes(globalFilter)) {
             if (sectionIndex === originalTask.sectionIndex) {
-                listItem = listItemCache;
+                const dateFromFileName = new Lazy(() => DateFallback.fromPath(originalTask.path));
```

Comment:
Unfortunately the code for LivePreview uses different file-updating code, and I suspect this will need a check too:

https://github.com/obsidian-tasks-group/obsidian-tasks/blob/5b0831c36a80c4cde2d64a6cd281bb4b51e9a142/src/LivePreviewExtension.ts#L83-L91

(Sorry about this. I have tried a few times to get my head around unifying the different file-writing behaviours in Tasks, unsuccessfully - it pre-dates my involvement in the project.)

Reply:
I think this could be in a different PR, I've updated the title of this one accordingly for a more specific scope...

Reply:
> I think this could be in a different PR, I've updated the title of this one accordingly for a more specific scope...

I'm happy for the Live Preview change to be in a different PR. Good idea to not hold up this improvement.

I've made the title a bit more user-focussed as it will appear in the release notes:\

> fix: Prevent rare overwrite of wrong line in Reading mode & results blocks

Reply:
Ah, I didn't know the PR titles are used directly in the release notes.

The new title I feel is not entirely accurate. 

Any sort of rare overwrite would be a timing issue, which this PR does not address.

This PR only addresses the issue in #688 and potentially #1659. At least for #688, this is 100% reproducible and has nothing to do with timing.

It could be the case that this PR unintentionally addresses any rare overwrites as well, but even so, does that justify the title?

Reply:
However, thinking more about this, maybe I misunderstand :)

Rare overwrite could also mean, "this is not a typical rendering of a task + checkbox and therefore this bug is rare, because these sorts of checkboxes won't be clicked often" ... ?

Reply:
Let's talk about this - it is too long and complicated to answer by typing...

Reply:
I'm happy with the PR title. I think you've added more content to the description - thank you!

I've added a little intro too... Hopefully that will clarify things for any users reviewing the change.